Ad9545 sys clock compensation

tachicialex opened the pull request:
Aux. DPLL can be used as closed loop compensation for the system clock.
A stable reference can be used as a source for the Aux. DPLL. This source
can be received at one of the Mx pins and routed through an aux. TDC
or can be an input Ref. X.

nunojsa commented on Documentation/devicetree/bindings/clock/clk-ad9545.yaml:
```diff
@@ -78,7 +81,63 @@ properties:
+      reg:
+        description: |
+          Address of the Auxiliary TDC.
+        maxItems: 1
```
this also looks that maximum is 1?

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -735,6 +793,85 @@ static int ad9545_parse_dt_ncos(struct ad9545_state *st)
+		}
+
+		if (addr > 1)
+			return -EINVAL;
```
maybe an error message here...

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -735,6 +793,85 @@ static int ad9545_parse_dt_ncos(struct ad9545_state *st)
+		}
+
+		if (val >= ARRAY_SIZE(ad9545_ref_m_clk_names))
+			return -EINVAL;
```
ditto...

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -735,6 +793,85 @@ static int ad9545_parse_dt_ncos(struct ad9545_state *st)
+
+	fwnode = dev_fwnode(st->dev);
+	fwnode_for_each_available_child_node(fwnode, child) {
+		if (!fwnode_property_present(child, "adi,pin-nr"))
```
I'm confused by this. Do we really want to continue in this case? It does not make much sense taking into account what we have some lines below

tachicialex replied:
Here i am doing DT node identification by checking for an unique property. I know it is bad, upstream complained about this, i am working on a way based on DT node name for a V2 upstream.

nunojsa replied:
I see. Good to know that this is being upstreamed. Naturally, we can do our internal review but I would then wait you have pushed this upstream to merge this PR. So that you can already sync this PR with upstream changes... Sounds good?

tachicialex replied:
Pushed this here because the hardware team needs this atm. for a project and because takes about a month to get a response from clock maintainers.

nunojsa replied:
I see, let's merge this then. Just make sure to sync things up when upstreamed...

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -1968,6 +2113,191 @@ static int ad9545_aux_ncos_setup(struct ad9545_state *st)
+	int i;
+
+	st->clks[AD9545_CLK_AUX_TDC] = devm_kzalloc(st->dev, ARRAY_SIZE(ad9545_aux_tdc_clk_names) *
+						    sizeof(struct clk *), GFP_KERNEL);
```
maybe `devm_kcalloc()`? also the `sizeof()` looks a bit odd... maybe something like `sizeof(*st->clks[AD9545_CLK_AUX_TDC])`?

nunojsa reviewed (commented):
Overall looks good. Also take a look at the checkpatch warning of the CI

tachicialex commented:
Changelog v1 -> v2:
- added maximum address to Aux. TDC in bindings
- added error messages in ad9545_parse_dt_tdcs
- devm_kzalloc to devm_kcalloc in ad9545_aux_tdcs_setup
- moved clock/ad9545.h changes in the bindings commit to supress warning
-

nunojsa reviewed (approved):
Looks good to me. Let me know if there's any hurry in merging this before being accepted upstream. Ideally, I would like to avoid "sync with upstream" PRs (when possible)
